### PR TITLE
4587 Correct Name to Numbered Company

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#4587

*Description of changes:*

* Implements the option to change to a numbered benefit company from a named company.

* Refactored name change event handler.

* Commit count is skewed from working on a branch from branch during dependency PR work. Please disregard :)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Review by vysakh-menon-aot — APPROVED** [review]

File: src/components/Company/CompanyName/CorrectNameToNumber.vue [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<template>
+  <v-form lazy-validation>
+    <v-layout row>
+        <v-checkbox
+          v-model="correctToNumbered"
+          label="Change the company name to 1234567 B.C. Ltd."
```
Comment (lekshmimallika-aot) [inline src/components/Company/CompanyName/CorrectNameToNumber.vue:6]: Is this a hardcoded value or is it supposed to be {identifier} B.C. Ltd.

Comment (severinbeauvais) [inline src/components/Company/CompanyName/CorrectNameToNumber.vue:6]: I know the mockup shows this, but maybe we should instead display "[Incorporation Number] B.C. Ltd."? Please check with Scott.

File: src/components/Company/CompanyName/CorrectNameToNumber.vue [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+@Component({})
+export default class CorrectNameToNumber extends Vue {
+  /** Form Submission Prop */
+  @Prop({ default: null }) formType: CorrectionTypes
+
+  @Action setNameRequest!: ActionBindingIF
+
+  @Getter getApprovedName!: string
+  @Getter getEntityType!: string
```
Comment (severinbeauvais) [inline src/components/Company/CompanyName/CorrectNameToNumber.vue:29]: Type? :)

**Review by severinbeauvais — APPROVED** [review]

File: src/components/Company/CompanyName/CorrectNameToNumber.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/Company/CompanyName/CorrectNameToNumber.vue:6]: I didn't even think of this until you guys mentioned it :) I have updated to be populated by the Companies Business identifier in the next commit!

Comment (cameron-freshworks) [reply src/components/Company/CompanyName/CorrectNameToNumber.vue:6]: Great points! I took the mock up literally hahah updated in next commit

File: src/components/Company/CompanyName/CorrectNameToNumber.vue (hunk 2 above)
Comment (cameron-freshworks) [reply src/components/Company/CompanyName/CorrectNameToNumber.vue:29]: fixed :)

**Review by lekshmimallika-aot — APPROVED** [review]
Looks good!

File: src/components/Company/CompanyName/CorrectCompanyName.vue [hunk 3]
```diff
@@ -59,16 +59,18 @@ export default class CorrectCompanyName extends Vue {
   /** Watch for form submission and emit results. */
   @Watch('formType')
   private async onSubmit (): Promise<any> {
+    console.log(this.formType)
```
Comment (severinbeauvais) [inline src/components/Company/CompanyName/CorrectCompanyName.vue:62]: Please remove this in a future commit.
Also the console.log on line 64.

Comment (cameron-freshworks) [reply src/components/Company/CompanyName/CorrectCompanyName.vue:62]: Shooooot...

Comment (cameron-freshworks) [reply src/components/Company/CompanyName/CorrectCompanyName.vue:62]: Will do

**Pull request merged**